core/src/components/modal/gestures/sheet.ts

```diff
@@ -462,6 +452,7 @@ export const createSheetGesture = (
         .onFinish(
           () => {
             if (shouldRemainOpen) {
+              swapFooterVisibility('original');
```

Comment:
Let's make sure to keep the comment and the check.

```suggestion
              /**
               * If expandToScroll is disabled, we need to swap
               * the footer visibility to the original, so if the modal
               * is dismissed, the footer dismisses with the modal
               * and doesn't stay on the screen after the modal is gone.
               */
              if (!expandToScroll) {
                swapFooterVisibility('original');
              }
```